### PR TITLE
Create new issue from code

### PR DESCRIPTION
**Pull request by rogerluo410**
Feat: add reference in new issue with permalink menu for code view.

For  feature: issue from code:
[](url)https://github.com/go-gitea/gitea/issues/6125

vdpna-4si3s

### CODE REVIEW

File: routers/repo/issue.go [hunk 1]
```diff
@@ -781,6 +781,11 @@ func NewIssue(ctx *context.Context) {
 	ctx.Data["TitleQuery"] = title
 	body := ctx.Query("body")
 	ctx.Data["BodyQuery"] = body
+
+	// get permalink query
+	permalink := ctx.Query("permalink")
+	ctx.Data["Permalink"] = permalink
```
Comment (zeripath) [inline routers/repo/issue.go]: Why don't you just use the body query parameter we already have here?

File: templates/repo/issue/comment_tab.tmpl [hunk 2]
```diff
@@ -4,8 +4,8 @@
 </div>
 <div class="field">
 	<div class="ui bottom active tab" data-tab="write">
-		<textarea id="content" class="edit_area js-quick-submit" name="content" tabindex="4" data-id="issue-{{.RepoName}}" data-url="{{.Repository.APIURL}}/markdown" data-context="{{.Repo.RepoLink}}">
-{{if .BodyQuery}}{{.BodyQuery}}{{else if .IssueTemplate}}{{.IssueTemplate}}{{else if .PullRequestTemplate}}{{.PullRequestTemplate}}{{else}}{{.content}}{{end}}</textarea>
+	 	<textarea id="content" class="edit_area js-quick-submit" name="content" tabindex="4" data-id="issue-{{.RepoName}}" data-url="{{.Repository.APIURL}}/markdown" data-context="{{.Repo.RepoLink}}">{{if .BodyQuery}}{{.BodyQuery}}{{else if .IssueTemplate}}{{.IssueTemplate}}{{else if .PullRequestTemplate}}{{.PullRequestTemplate}}{{else if .Permalink}}{{.Permalink}}{{else}}{{.content}}{{end}}
```
Comment (zeripath) [inline templates/repo/issue/comment_tab.tmpl]: If you used the body query parameter you wouldn't need this change

File: routers/repo/view.go [hunk 3]
```diff
@@ -704,6 +704,7 @@ func renderCode(ctx *context.Context) {
 	ctx.Data["TreeLink"] = treeLink
 	ctx.Data["TreeNames"] = treeNames
 	ctx.Data["BranchLink"] = branchLink
+	ctx.Data["AppURLTmp"] = strings.TrimRight(setting.AppURL, "/")
```
Comment (zeripath) [inline routers/repo/view.go]: I really think you should avoid this if at all possible

File: templates/repo/view_file.tmpl [hunk 4]
```diff
@@ -111,6 +111,15 @@
 						{{end}}
 					</tbody>
 				</table>
+				<div class="code-view-menu-list ui fluid popup transition hidden">
+					<div class="ui column relaxed equal height">
+						<div class="column">
+							<div class="ui link list">
+								<a class="item ref-in-new-issue" href="{{.RepoLink}}/issues/new?permalink={{.AppURLTmp}}{{.RepoLink}}/src/commit/{{.CommitID}}/{{EscapePound .TreePath}}">{{.i18n.Tr "repo.issues.context.reference_issue"}}</a>
```
Comment (zeripath) [inline templates/repo/view_file.tmpl]: There has to be a better way of getting the URL to the repo than doing this

{{.AppURLTmp}}{{.RepoLink}}


I'll have a look.

Also use the body query parameter.

Comment (noerw) [reply templates/repo/view_file.tmpl]: `{{AppUrl}}{{AppSubUrl}}{{.RepoLink}}` is what you're looking for, these are global template helpers.
Though I see the issue that AppUrl has a trailing `/` and .RepoLink has a leading `/`.. so more like
```
{{URLJoin AppUrl AppSubUrl .RepoLink}}
```

Comment (rogerluo410) [reply templates/repo/view_file.tmpl]: Thx, fixed.

File: templates/repo/issue/comment_tab.tmpl [hunk 5]
```diff
@@ -4,8 +4,8 @@
 </div>
 <div class="field">
 	<div class="ui bottom active tab" data-tab="write">
-		<textarea id="content" class="edit_area js-quick-submit" name="content" tabindex="4" data-id="issue-{{.RepoName}}" data-url="{{.Repository.APIURL}}/markdown" data-context="{{.Repo.RepoLink}}">
-{{if .BodyQuery}}{{.BodyQuery}}{{else if .IssueTemplate}}{{.IssueTemplate}}{{else if .PullRequestTemplate}}{{.PullRequestTemplate}}{{else}}{{.content}}{{end}}</textarea>
+	 	<textarea id="content" class="edit_area js-quick-submit" name="content" tabindex="4" data-id="issue-{{.RepoName}}" data-url="{{.Repository.APIURL}}/markdown" data-context="{{.Repo.RepoLink}}">{{if .BodyQuery}}{{.BodyQuery}}{{else if .IssueTemplate}}{{.IssueTemplate}}{{else if .PullRequestTemplate}}{{.PullRequestTemplate}}{{else}}{{.content}}{{end}}
+	  	</textarea>
```
Comment (zeripath) [inline templates/repo/issue/comment_tab.tmpl]: Let's fix the indenting here:

```suggestion
		<textarea id="content" class="edit_area js-quick-submit" name="content" tabindex="4" data-id="issue-{{.RepoName}}" data-url="{{.Repository.APIURL}}/markdown" data-context="{{.Repo.RepoLink}}">
			{{- if .BodyQuery}}{{.BodyQuery}}{{else if .IssueTemplate}}{{.IssueTemplate}}{{else if .PullRequestTemplate}}{{.PullRequestTemplate}}{{else}}{{.content}}{{end -}}
		</textarea>
```

File: templates/repo/view_file.tmpl [hunk 6]
```diff
@@ -111,6 +111,15 @@
 						{{end}}
 					</tbody>
 				</table>
+				<div class="code-view-menu-list ui fluid popup transition hidden">
+					<div class="ui column relaxed equal height">
+						<div class="column">
+							<div class="ui link list">
+								<a class="item ref-in-new-issue" href="{{.RepoLink}}/issues/new?body={{URLJoin AppUrl AppSubUrl .RepoLink}}/src/commit/{{.CommitID}}/{{EscapePound .TreePath}}">{{.i18n.Tr "repo.issues.context.reference_issue"}}</a>
```
Comment (zeripath) [inline templates/repo/view_file.tmpl]: Don't just AppUrl to AppSubUrl - AppSubUrl is already in AppUrl - (I appreciate it's confusing)

```suggestion
								<a class="item ref-in-new-issue" href="{{.RepoLink}}/issues/new?body={{URLJoin AppUrl .RepoLink}}/src/commit/{{.CommitID}}/{{EscapePound .TreePath}}">{{.i18n.Tr "repo.issues.context.reference_issue"}}</a>
```

**Review by zeripath — COMMENTED** [review]
almost there! Thanks for your changes.

**Review by zeripath — APPROVED** [review]

**Review by noerw — APPROVED** [review]
thank you!

**Review by lafriks — APPROVED** [review]

**Comment by techknowlogick** [comment]
🚀

**Pull request merged**